extensions/amp-experiment/1.0/mutation-parser.js

```diff
@@ -29,48 +30,55 @@ const MUTATION_TYPES = ['attributes', 'characterData', 'childList'];
 
 /**
  * Function to find all selectors of the mutation
- * and return a function to apply the identified
- * MutationRecord
+ * and return an object with the parsed mutation record
  * @param {!JsonObject} mutation
  * @param {!Document} document
- * @return {!Function}
+ * @return {!Object}
  */
 export function parseMutation(mutation, document) {
   const mutationRecord = assertMutationRecordFormat(mutation);
 
   const stringifiedMutation = JSON.stringify(mutation);
 
-  setSelectorToElement(mutationRecord, document);
+  setSelectorToElements(mutationRecord, document);
+
+  mutationRecord.mutations = [];
 
   if (mutationRecord['type'] === 'attributes') {
     assertAttributeMutationFormat(mutationRecord, stringifiedMutation);
 
-    // Get the AttribeMutationEntry
-    // From the Attribute allow list
-    const allowedAttributeMutationEntry = getAllowedAttributeMutationEntry(
-      mutationRecord,
-      stringifiedMutation
-    );
-
-    // Assert the attribute mutation passes it's check
-    // that is allowed and validated.
-    userAssert(
-      allowedAttributeMutationEntry.validate(mutationRecord),
-      'Mutation %s has an an unsupported value.',
-      stringifiedMutation
-    );
-
-    // Return the corresponding mutation
-    // For the allowed attribute mutation
-    return allowedAttributeMutationEntry.mutate.bind(this, mutationRecord);
+    mutationRecord.targetElements.forEach(element => {
+      // Get the AttribeMutationEntry
+      // From the Attribute allow list
+      const allowedAttributeMutationEntry = getAllowedAttributeMutationEntry(
+        mutationRecord,
```

Comment:
I have an idea. Besides the JsonObject `mutationRecord`, we introduce a new object type call `mutationRecordDef` which looks like
```
{
  element: ..,
  type:...
 ...
}
```
The idea is we create multiple `mutationRecordDef` Object from one `mutationRecord` JsonObject. In this way we can make minimum change to the `AttributeMutationEntry` class. Let me know what do you think about this approach?

Reply:
Yeah, I definitely agree with the approach, but not the naming 😂 

Since I'll just be doing Roadshow tommorow, want to bikeshed the name when I get back? I think we need to be clear of JSON vs. Object Mutation Records. So I'll think about it in my free time.

Thanks! 😄

Reply:
Oh! I agree it's a bad name 😂. 

> I think we need to be clear of JSON vs. Object Mutation Records

+100 to that